### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by Globulopolis**
Fixed bug with cache update in comments.
Complete rewrites for `about` and `comment(s)` views.
Fixed bugs in blacklist view.
Fixed bug in admin with access checking.
Change old class names to new.
Applied Joomla code style.
Small code cleanup.

!NB All other views w/o changes and may not working as expected.

### CODE REVIEW

File: administrator/models/modellist.php [hunk 1]
```diff
@@ -176,16 +176,6 @@ protected function _getListCount($query)
 		return $db->getNumRows();
 	}
 
-	protected function canDelete($record)
```
Comment (exstreme) [inline administrator/models/modellist.php:179]: Why you deleted this functions? 
It is more practical and no code duplication

Comment (exstreme) [reply administrator/models/modellist.php:179]: and I don't like it when there is a lot of nesting, the code becomes unreadable and incomprehensible. I will give an example of how I would do it, if you like it, you can take a note.
```
   public function checkin($pks = array())
    {
        $pks     = (array) $pks;
        $table   = $this->getTable('Comment', 'JCommentsTable');
        $checkin = property_exists($table, 'checked_out');
        $count   = 0;

        if($checkin === false) {
            return $count;
        }

        foreach ($pks as $pk)
        {
            if (!$table->load($pk) || !$table->checkin($pk)) {
                $this->setError($table->getError());

                return false;
            }
            
            if ($table->checked_out > 0)
            {
                $count++;
            }
        }
        

        return $count;
    }
```

**Review by exstreme — CHANGES_REQUESTED** [review]

File: administrator/models/modellist.php (hunk 1 above)
Comment (Globulopolis) [reply administrator/models/modellist.php:179]: > 
> 
> Why you deleted this functions? It is more practical and no code duplication

This code used only in two models. And in these two models only in one place it's used. This is no code dupes.

Comment (Globulopolis) [reply administrator/models/modellist.php:179]: > 
> 
> and I don't like it when there is a lot of nesting, the code becomes unreadable and incomprehensible. I will give an example of how I would do it, if you like it, you can take a note.
> 
> ```
>    public function checkin($pks = array())
>     {
>         $pks     = (array) $pks;
>         $table   = $this->getTable('Comment', 'JCommentsTable');
>         $checkin = property_exists($table, 'checked_out');
>         $count   = 0;
> 
>         if($checkin === false) {
>             return $count;
>         }
> 
>         foreach ($pks as $pk)
>         {
>             if (!$table->load($pk) || !$table->checkin($pk)) {
>                 $this->setError($table->getError());
> 
>                 return false;
>             }
>             
>             if ($table->checked_out > 0)
>             {
>                 $count++;
>             }
>         }
>         
> 
>         return $count;
>     }
> ```

The code for checking is not mine. I just apply a code style. In next update I will use you code and replace. Tanks for suggestion.

**Pull request merged**